home-assistant-plugin/custom_components/xknx/sensor.py

```diff
@@ -13,21 +14,27 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     entities = []
     for device in hass.data[DOMAIN].xknx.devices:
         if isinstance(device, XknxSensor):
-            entities.append(KNXSensor(device))
+            entities.append(KNXSensor(device, hass))
     async_add_entities(entities)
 
 
 class KNXSensor(KnxEntity, Entity):
     """Representation of a KNX sensor."""
 
-    def __init__(self, device: XknxSensor):
+    def __init__(self, device: XknxSensor, hass: HomeAssistant):
         """Initialize of a KNX sensor."""
+        if device.value_template is not None:
+            device.value_template.hass = hass
         super().__init__(device)
 
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._device.resolve_state()
+        state = self._device.resolve_state()
+        template = self._device.value_template
+        if template is not None and state is not None:
+            state = template.async_render({"value": state})
```

Comment:
```suggestion
        if template is not None and state is not None:
            try:
                state = template.async_render({"value": state})
            except TemplateError as ex:
                # handle error
```
Not sure how to handle the error here. Maybe return `None` and log something... you can get inspiration here https://github.com/home-assistant/core/search?q=except+TemplateError 😉